manager/controlapi/secret.go

```diff
@@ -31,6 +31,34 @@ func secretFromSecretSpec(spec *api.SecretSpec) *api.Secret {
 	}
 }
 
+// checkSecretInUse does a best effort to find if the passed in secret is in
+// use by any services
+func (s *Server) checkSecretInUse(secretID string) error {
+	var (
+		services []*api.Service
+		err      error
+	)
+
+	s.store.View(func(tx store.ReadTx) {
```

Comment:
This function is racy: by the time it returns true, the result may not actually be true and vice versa. The transaction needs to be held for the duration of the action take based upon the result.